src/main/java/org/broadinstitute/hellbender/utils/config/GATKConfig.java

```diff
@@ -0,0 +1,146 @@
+package org.broadinstitute.hellbender.utils.config;
+
+import org.aeonbits.owner.Accessible;
+import org.aeonbits.owner.Mutable;
+import org.aeonbits.owner.Config.LoadPolicy;
+import org.aeonbits.owner.Config.LoadType;
+import org.aeonbits.owner.Config.Sources;
+
+import java.util.List;
+
+/**
+ * Configuration file for GATK options.
+ */
+@LoadPolicy(LoadType.MERGE)
+@Sources({
+        "file:${" + GATKConfig.CONFIG_FILE_VARIABLE_NAME + "}",                         // Variable for file loading
+        "file:GATKConfig.properties",                                                   // Default path
+        "classpath:org/broadinstitute/hellbender/utils/config/GATKConfig.properties"    // Class path
+})
+public interface GATKConfig extends Mutable, Accessible {
```

Comment:
@magicDGS I'm not sure what you mean by (1).

2. Extending GATKConfig to use more properties is possible.  The config factory cues off the class type you give it to pull a config object out of the cache (or create one), so the scenario you describe won't work.  I'll need to add in an abstraction for which specific configuration class to use at runtime.  A downstream user will then be able to specify that class and it should work as expected.

3. I added in the ability to load a separate configuration file on the command-line with an argument.  The Sources annotation controls the search path for the configuration file.  When defining a new configuration class you can change the Sources annotation to look in multiple places in a hierarchical fashion (as well as whether or not you want to merge configurations from these sources or take them each as a whole - we default to merging so you can override some of the values, but not all of them).